scripts/omni_run.py

```diff
@@ -2,220 +2,184 @@
             else: continue
-
-    for ip, pol in enumerate(uvd.polarization_array):
-        pol = aipy.miriad.pol2str[pol]
-        if pol != opts.pol: 
-            continue
-        for nbl, (i,j) in enumerate(map(uvd.baseline_to_antnums, uvd.baseline_array[:uvd.Nbls])):
-            if not (i, j) in bls and not (j, i) in bls:
-                continue
-            d[(i,j)] = {pol: uvd.data_array.reshape(uvd.Ntimes, uvd.Nbls, uvd.Nspws, uvd.Nfreqs, uvd.Npols)[:, nbl, 0, :, ip]}
-            f[(i,j)] = {pol: np.logical_not(uvd.flag_array.reshape(uvd.Ntimes, uvd.Nbls, uvd.Nspws, uvd.Nfreqs, uvd.Npols)[:, nbl, 0, :, ip])}
     
+    ## read data into dictionaries
+    d,f = {},{}
+    for ip,pp in enumerate(pols):
+        uvdp = uvd_dict[pp]
+        
+        if ip==0:
+            for nbl, (i,j) in enumerate(map(uvdp.baseline_to_antnums, uvdp.baseline_array[:uvdp.Nbls])):
+                if not (i, j) in bls and not (j, i) in bls: continue
+                d[(i,j)] = {}
+                f[(i,j)] = {}
+        
+        #XXX I *really* don't like looping again, but I'm not sure how better to do it
+        for nbl, (i,j) in enumerate(map(uvdp.baseline_to_antnums, uvdp.baseline_array[:uvdp.Nbls])):
+            if not (i, j) in bls and not (j, i) in bls: continue
+            d[(i,j)][pp] = uvdp.data_array.reshape(uvdp.Ntimes, uvdp.Nbls, uvdp.Nspws, uvdp.Nfreqs, uvdp.Npols)[:, nbl, 0, :, 0]
+            f[(i,j)][pp] = np.logical_not(uvdp.flag_array.reshape(uvdp.Ntimes, uvdp.Nbls, uvdp.Nspws, uvdp.Nfreqs, uvdp.Npols)[:, nbl, 0, :, 0])
+    
+    ## Finally prepared to run omnical
     print '   Running Omnical'
     m2, g3, v3 = run_omnical(d, info, gains0=g0)
-
-    # need wgts for xtalk
+    
+    ## Collect weights for xtalk
     wgts, xtalk = {}, {}
-    for p in pols:
-        wgts[p] = {}  # weights dictionary by pol
+    for pp in pols:
+        wgts[pp] = {}  # weights dictionary by pol
         for i,j in f:
             if (i,j) in bls:
-                wgts[p][(i, j)] = np.logical_not(f[i,j][p]).astype(np.int)
+                wgts[pp][(i, j)] = np.logical_not(f[i,j][pp]).astype(np.int)
             else:  # conjugate
-                wgts[p][(j, i)] = np.logical_not(f[i,j][p]).astype(np.int)
-
-    xtalk = compute_xtalk(m2['res'], wgts)  # xtalk is time-average of residual
-    m2['history'] = 'OMNI_RUN: ' + ''.join(sys.argv) + '\n'
+                wgts[pp][(j, i)] = np.logical_not(f[i,j][pp]).astype(np.int)
+    # xtalk is time-average of residual: data - omnical model
+    xtalk = compute_xtalk(m2['res'], wgts)  
+    
+    ## Append metadata parameters 
+    m2['history'] = 'OMNI_RUN: '+' '.join(sys.argv) + '\n'
     m2['times'] = t_jd
     m2['lsts'] = t_lst
     m2['freqs'] = freqs
-    m2['inttime'] = uvd.integration_time
-    optional = {'observer': 'Zaki Ali (zakiali@berkeley.edu)'}
+    m2['inttime'] = t_int
+    optional = {'observer': 'obs (example@institution.edu)'}
```

Comment:
Sorry, I'm not sure what you're referring to. Do you mean to rename `obs` as `heracal`, or to leave our the email altogether? Or both?